src/cargo/util/progress.rs

```diff
@@ -39,6 +40,25 @@ struct Format {
     max_print: usize,
 }
 
+enum ProgressCode {
```

Comment:
This is Windows-only feature, right? Is it possible to add `#[cfg(windows)]` for all relevant code and items?

Also, having doc comments on each new items would be fantastic. Thanks!